File: El problema de Josephus/El problema de Josephus.py
#define Hd

def answr(n:int) -> int:
    if n<2: return 1
    return ((2*answr(n/2)+1) if n&1 else 2*answr(n/2)-1)

t : int
t = int(input())
while t:
    t-=1
    n = int(input())
    print(answr(n))

File: El problema de Josephus/El problema de Josephus.cpp
#include<stdio.h>
#define Hd main()

long long answr(long long n){
    if(n<=1)return 1;
    return (n%2==0? 2*answr(n/2)-1:2*answr(n/2)+1);
}

Hd
{
    long long t;
    scanf("%lld",&t);
    while(t--){
        long long n;
        scanf("%lld",&n);
        long long a=answr(n);
        printf("%lld\n",a);
    }
}
